public/app/features/variables/editor/VariableEditorListRow.tsx

```diff
@@ -0,0 +1,171 @@
+}: VariableEditorListRowProps): ReactElement {
+  const styles = useStyles2(getStyles);
+  const definition = getDefinition(variable);
+  const usages = getVariableUsages(variable.id, usageTree);
+  const passed = usages > 0 || isAdHoc(variable);
+  const identifier = toVariableIdentifier(variable);
+  return (
+    <Draggable draggableId={JSON.stringify(identifier)} index={index}>
+      {(provided) => (
+        <tr ref={provided.innerRef} {...provided.draggableProps}>
+          <td className={styles.column}>
+            <span
+              onClick={(event) => {
+                event.preventDefault();
+                propsOnEdit(identifier);
+              }}
+              className={styles.nameLink}
+              aria-label={selectors.pages.Dashboard.Settings.Variables.List.tableRowNameFields(variable.name)}
+            >
+              {variable.name}
+            </span>
+          </td>
+          <td
+            className={styles.definitionColumn}
+            onClick={(event) => {
+              event.preventDefault();
+              propsOnEdit(identifier);
+            }}
+            aria-label={selectors.pages.Dashboard.Settings.Variables.List.tableRowDefinitionFields(variable.name)}
+          >
+            {definition}
+          </td>
+
+          <td className={styles.column}>
+            <VariableCheckIndicator passed={passed} />
+          </td>
+
+          <td className={styles.column}>
+            <VariableUsagesButton id={variable.id} isAdhoc={isAdHoc(variable)} usages={usagesNetwork} />
+          </td>
+
+          <td className={styles.column}>
+            <IconButton
+              onClick={(event) => {
+                event.preventDefault();
+                propsOnDuplicate(identifier);
+              }}
+              name="copy"
+              title="Duplicate variable"
+              aria-label={selectors.pages.Dashboard.Settings.Variables.List.tableRowDuplicateButtons(variable.name)}
+            />
+          </td>
+
+          <td className={styles.column}>
+            <IconButton
+              onClick={(event) => {
+                event.preventDefault();
+                propsOnDelete(identifier);
+              }}
+              name="trash-alt"
+              title="Remove variable"
+              aria-label={selectors.pages.Dashboard.Settings.Variables.List.tableRowRemoveButtons(variable.name)}
+            />
+          </td>
+          <td className={styles.column}>
+            <div {...provided.dragHandleProps} className={styles.dragHandle}>
+              <Icon name="draggabledots" size="lg" />
+            </div>
+          </td>
+        </tr>
+      )}
+    </Draggable>
+  );
+}
+
+function getDefinition(model: VariableModel): string {
+  let definition = '';
+  if (isQuery(model)) {
+    definition = model.definition ? model.definition : typeof model.query === 'string' ? model.query : '';
+  } else if (hasOptions(model)) {
+    definition = model.query;
+  }
```

Comment:
maybe this might be a bit easier to read if it was over multiple ifs instead of the nested ternaries?

Reply:
Great suggestion, I'll push a fix soon!

Reply:
Fixed!